### PR TITLE
Employee status change on leave

### PR DESCRIPTION
**Pull request by yusuffGoodie**
## Is this a Feature, Chore or Bug?
- [*] Feature
- [] Chore
- [] Bug

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.

## Is there a business logic within a doctype?
    - [*] Yes
    - [] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [*] Existing Data
- [] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [*] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [*] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [*] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/utils.py [hunk 1]
```diff
@@ -3640,4 +3640,45 @@ def send_work_anniversary_reminders():
...
+
+    # Get a list of employees with approved leave applications
+    employees_on_leave = [leave['employee'] for leave in leave_applications]
+
+    if not employees_on_leave:
+        frappe.log_error(_("No employees found with approved leave applications for today or earlier."))
+        return
+
+    # Fetch all employees with 'Active' status
+    active_employees = frappe.get_all('Employee',
```
Comment (NoBoneZ) [inline one_fm/utils.py]: You could filter the employees on leave in the filter, please

File: one_fm/utils.py [hunk 2]
```diff
@@ -3640,4 +3640,45 @@ def send_work_anniversary_reminders():
...
+    # Fetch all employees with 'Active' status
+    active_employees = frappe.get_all('Employee', 
+        filters={
+            'status': 'Active'
+        }, 
+        fields=['name']
+    )
+
+    # Filter active employees who are in the leave applications list
+    employees_to_set_vacation = [
```
Comment (NoBoneZ) [inline one_fm/utils.py:3671]: so this would not be necessary anymore

**Review by NoBoneZ — CHANGES_REQUESTED** [review]
@yusuffAFEX , kindly check the comments

File: one_fm/utils.py (hunk 2 above)
Comment (yusuffGoodie) [reply one_fm/utils.py:3671]: DOne

File: one_fm/utils.py [hunk 3]
```diff
@@ -3640,4 +3640,38 @@ def send_work_anniversary_reminders():
...
+    current_date = today()
+
+    # Fetch approved leave applications where `from_date` is today or earlier, along with the employee's status
+    leave_applications = frappe.get_all('Leave Application', 
+        filters={
+            'status': 'Approved',
+            'from_date': ['<=', current_date],
+            'to_date': ['>', current_date]
+        },
+        fields=['employee', 'from_date', 'employee.status']  # Fetch employee status directly
```
Comment (NoBoneZ) [inline one_fm/utils.py]: @yusuffAFEX , you are collecting , the "from_date" field, but you are not using it, also, if that employee.status, would work, also you are not using the field

File: one_fm/utils.py (hunk 2 above)
Comment (NoBoneZ) [reply one_fm/utils.py:3671]: also, you still need to filter out inactive employees, 

so you need to get a list of name(id) of employee on leave, then also those that are active,

Comment (yusuffGoodie) [reply one_fm/utils.py:3671]: I don't understand this. Why do I need filter out inactive

Comment (NoBoneZ) [reply one_fm/utils.py:3671]: @yusuffAFEX , so it is possible to have  employees with a different status from active, and the aim of this method is to set those with active status, to vacation, and let's assume, an employee is already set to vacation,  there would be no need to  set it again

File: one_fm/utils.py (hunk 3 above)
Comment (yusuffGoodie) [reply one_fm/utils.py]: Okay

File: one_fm/utils.py (hunk 2 above)
Comment (yusuffGoodie) [reply one_fm/utils.py:3671]: Yes, that's exactly what the list comprehension is doing. It is filter those who that have active status to set them to vacation.

Comment (NoBoneZ) [reply one_fm/utils.py:3671]: so, here, it is leave status that is being checked, not the employee status
image

Comment (yusuffGoodie) [reply one_fm/utils.py:3671]: It is employee status. I tested it

Comment (NoBoneZ) [reply one_fm/utils.py:3671]: Oh, great great, thank you very much

**Review by NoBoneZ — APPROVED** [review]

**Pull request merged**